src/posh/plugin_base.cljc

```diff
@@ -99,19 +99,20 @@
                  query-result         (:results (get (:cache posh-atom-with-query) storage-key))
                  query-ratom          (or (get (:ratoms posh-atom-with-query) storage-key)
                                           ((:ratom dcfg) query-result))
-                 query-reaction       ((:make-reaction dcfg)
-                                        (fn []
-                                          ;;(println "RENDERING: " storage-key)
-                                          @query-ratom)
-                                        :on-dispose
-                                        (fn [_ _]
-                                          ;;(println "no DISPOSING: " storage-key)
-                                          (when-not (= (:cache options) :forever)
-                                            (swap! posh-atom
-                                                   (fn [posh-atom-val]
-                                                     (assoc (p/remove-item posh-atom-val storage-key)
-                                                       :ratoms (dissoc (:ratoms posh-atom-val) storage-key)
-                                                       :reactions (dissoc (:reactions posh-atom-val) storage-key)))))))]
+                 query-reaction       ((:derive-reaction dcfg) [query-ratom] storage-key (fn [q] q))]
+                ;  query-reaction       ((:make-reaction dcfg)
```

Comment:
I haven't thought through what would happen if you change the implementation of `query-reaction` to what you have now, but assuming your version works, could you remove the commented portion of code?